totalRP3/modules/register/main/register_exchange.lua

```diff
@@ -484,63 +484,66 @@ end
 
 local UNIT_TOKENS = { "target", "mouseover", "player", "focus" };
 UNIT_TOKENS = tInvert(UNIT_TOKENS);
-TRP3_API.slash.registerCommand({
-	id = "open",
-	helpLine = " " .. loc.PR_SLASH_OPEN_HELP,
-	handler = function(...)
-		local args = {...};
 
-		if commandOpeningTimerHandle then
-			commandOpeningTimerHandle:Cancel();
-		end
+function TRP3_API.slash.openProfile(...)

```

Comment:
As you said, this function should indeed be moved, maybe to the register_main file. Split the actual fetching and opening of the profile and indicate in the function's name that a request will be sent, so name it something like `TRP3_API.register.getAndOpenProfile(charID)` and only have the actual slash command code in this file.